src/app/ApplicationTemplate.Tests.Functional/KillSwitch/KillSwitchShould.cs

```diff
@@ -12,6 +12,7 @@ namespace KillSwitch;
 /// <summary>
 /// Tests for the kill switch flow.
 /// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1001:Types that own disposable fields should be disposable", Justification = "FunctionalTestBase provides a DisposeAsync method that is automatically called.")]
```

Comment:
You don't need the async aspect of DisposeAsync so you could simply implement IDiposable on this specific class instead of making DisposeAsunc virtual.